src/pudl_archiver/archivers/eia/eiarecs.py

```diff
@@ -0,0 +1,280 @@
+            "state-data": defaultdict(lambda: self.__get_tab_links),
+        }
+
+        # Add the exceptions - skip the 2009 and 2015 methodology sections for now
+        tab_handlers["methodology"][2015] = self.__skip
+        tab_handlers["methodology"][2009] = self.__skip
+
+        zip_path = self.download_directory / f"eia-recs-{year}.zip"
+        paths_within_archive = []
+        for tab in tab_infos:
+            paths_within_archive += await tab_handlers[tab.name][tab.year](
+                tab_info=tab, zip_path=zip_path
+            )
+
+        self.logger.info(f"Looking for original forms for {year}")
+        original_forms_within_archive = await self.__get_original_forms(year, zip_path)
+
+        self.logger.info(f"Got original forms for {year}, returning ResourceInfo list.")
+        return ResourceInfo(
+            local_path=zip_path,
+            partitions={"year": year},
+            layout=ZipLayout(
+                file_paths=paths_within_archive + original_forms_within_archive
+            ),
+        )
+
+    async def __add_links_to_archive(
+        self, url_paths: dict[str, str], zip_path: Path
+    ) -> list[str]:
+        """Download and add link contents to a zipfile.
+
+        Skips links that lead to HTML content since these are usually broken links.
+
+        Args:
+            url_paths: mapping from URLs to the filenames we want them to have
+                in the zip.
+            zip_path: path to the archive
+
+        Returns:
+            list[str]: the filepaths, relative to the archive root, that we
+                just added.
+        """
+        data_paths_in_archive = []
+        for link, output_filename in url_paths.items():
+            download_path = self.download_directory / output_filename
+            logger.debug(f"Fetching {link} to {download_path}")
+            await self.download_file(link, download_path, timeout=120)
+            with download_path.open("rb") as f:
+                # TODO 2025-02-04: check html-ness against the suffix... if we
+                # have a php/html/cfm/etc. we probably actually *do* want the
+                # html file.
+                if self.__is_html_file(f):
+                    logger.info(f"{link} was HTML file - skipping.")
+                    continue
+                self.add_to_archive(
+                    zip_path=zip_path,
+                    filename=output_filename,
+                    blob=f,
+                )
+                logger.debug(f"Added {link} to {zip_path} as {output_filename}")
+                data_paths_in_archive.append(output_filename)
+            download_path.unlink()
+        return data_paths_in_archive
+
+    async def __get_tab_links(self, tab_info: TabInfo, zip_path: Path) -> list[str]:
+        """Get the data files for a single tab."""
+        soup = await self.__get_soup(tab_info.url)
+        links_in_tab = soup.select("div.tab-contentbox a[href]")
```

Comment:
Sometimes the EIA page is actually missing links - e.g. in 1997, there are a bunch of files that exist on their server, but the links on the data display page don't point at them where they should:

* [C&E Top 4 states - appliance expenditures](https://www.eia.gov/consumption/residential/data/1997/pdf/four-states/appl_exp4states.pdf)
* [HC5 - appliances (households, not percents)](https://www.eia.gov/consumption/residential/data/1997/pdf/housing-characteristics/appl_household.pdf)
* [microdata questionnaire section a](https://www.eia.gov/consumption/residential/data/1997/pdf/puquex_a.pdf)

Do you think it's worth adding override logic for 1997? Personally leaning no.